[Brave News]: Tweaks round 5

fallaciousreasoning opened the pull request:
<!-- Add brave-browser issue below that this PR will resolve -->
Resolves https://github.com/brave/brave-browser/issues/35193
Resolves https://github.com/brave/brave-browser/issues/35235
Resolves https://github.com/brave/brave-browser/issues/35213

- Add sidebar menu on small screens
- Apply channel migrations to article channels (only affects UI)
- Sidebar font contrast improvements
- Sidebar icon only add buttons

[Screencast from 2024-01-11 11-43-54.webm](https://github.com/brave/brave-core/assets/7678024/4bad6962-6f99-4d4a-bf80-be0de6bea4f1)


<!-- CI-related labels that can be applied to this PR:
* CI/run-audit-deps (1) - check for known npm/cargo vulnerabilities (audit_deps)
* CI/run-network-audit (1) - run network-audit
* CI/run-upstream-tests - run Chromium unit and browser tests on Linux and Windows (otherwise only on Linux)
* CI/run-linux-arm64, CI/run-macos-arm64, CI/run-windows-arm64, CI/run-windows-x86 - run builds that would otherwise be skipped
* CI/skip - do not run CI builds (except noplatform)
* CI/skip-linux-x64, CI/skip-android, CI/skip-macos-x64, CI/skip-ios, CI/skip-windows-x64 - skip CI builds for specific platforms
* CI/skip-upstream-tests - do not run Chromium unit, or browser tests (otherwise only on Linux)
* CI/skip-all-linters - do not run presubmit and lint checks
* CI/storybook-url (1) - deploy storybook and provide a unique URL for each build

(1) applied automatically when some files are changed (see: https://github.com/brave/brave-core/blob/master/.github/labeler.yml)
-->

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://build-artifacts.brave.com/brave-browser-build-pr/bn-tweaks-5/4d3adb4fd961d7f95b5aa1dff7c82d3ec0db9287/storybook/index.html)

simonhong commented:
IMO, it would be great to make buttons container bg color more opaque when it's overlapped.
![image](https://github.com/brave/brave-core/assets/6786187/c08f357f-d7af-4f83-9eef-34fcc672bd7b)

simonhong reviewed (approved):
++ 👍🏼

sangwoo108 commented on components/brave_new_tab_ui/components/default/braveNews/SidebarMenu.tsx:
```diff
@@ -0,0 +1,57 @@
+    <Icon name="hamburger-menu" />
+    <Container ref={dialogRef as any} onClick={e => {
+      // Close the menu on click outside.
+      const bounds = e.currentTarget.getBoundingClientRect()
```
This is interesting technic using `fit-content` and `box-sizing: border-box`!

sangwoo108 reviewed (approved):
LGTM 👍

fallaciousreasoning commented:
@simonhong @sangwoo108 could you take another look - @martin-c0des updated the design a bit so I've:
1. Added a top bar on small screens
2. Changed the contrast on the Sidebar text
3. Changed the `Add` buttons on the Sidebar to be icon only buttons

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://build-artifacts.brave.com/brave-browser-build-pr/bn-tweaks-5/2921aa9ebe6ac96affa04b94371cd1ef2023b523/storybook/index.html)

simonhong commented on components/brave_new_tab_ui/tsconfig.json:
```diff
@@ -5,4 +5,4 @@
     "**/*.d.ts",
     "../definitions/*.d.ts"
-}
+, "../brave_news/browser/resources/SettingsButton.tsx"}
```
this change should be checked again? json seems broken also.

fallaciousreasoning replied:
Whoops, no! That was added by a move from VSCode. Nice catch!

simonhong reviewed (approved):
++ much better :)

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://build-artifacts.brave.com/brave-browser-build-pr/bn-tweaks-5/50953a60a02c4160dc31fd8deaa9c2980be784a1/storybook/index.html)

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://build-artifacts.brave.com/brave-browser-build-pr/bn-tweaks-5/cb7ebc00d2d1988bf4cc8388ee499b95befbb7a2/storybook/index.html)

mkarolin reviewed (approved):
`strings`++

stephendonner commented:
## Verification `IN-PROGRESS` using 

```


```



Resolves https://github.com/brave/brave-browser/issues/35193
Resolves https://github.com/brave/brave-browser/issues/35235
Resolves https://github.com/brave/brave-browser/issues/35213
